include/spl/sys/debug.h

```diff
@@ -63,22 +63,59 @@ void spl_dumpstack(void);
 	    spl_panic(__FILE__, __FUNCTION__, __LINE__,			\
 	    "%s", "VERIFY(" #cond ") failed\n"))
 
-#define	VERIFY3_IMPL(LEFT, OP, RIGHT, TYPE, FMT, CAST)	do {		\
-		TYPE _verify3_left = (TYPE)(LEFT);				\
-		TYPE _verify3_right = (TYPE)(RIGHT);				\
+#define	VERIFY3B(LEFT, OP, RIGHT)	do {				\
+		boolean_t _verify3_left = (boolean_t)(LEFT);		\
+		boolean_t _verify3_right = (boolean_t)(RIGHT);		\
 		if (!(_verify3_left OP _verify3_right))			\
 		    spl_panic(__FILE__, __FUNCTION__, __LINE__,		\
-		    "VERIFY3(" #LEFT " " #OP " " #RIGHT ") "		\
-		    "failed (" FMT " " #OP " " FMT ")\n",		\
-		    CAST (_verify3_left), CAST (_verify3_right));	\
+		    "VERIFY3(" #LEFT  #OP  #RIGHT ") "			\
```

Comment:
Let's preserve the space around the operator for readability, here and in the other cases. Take the following VERIFY as an example:

```c
VERIFY3S(error, !=, 0);
```

This PR outputs

```
VERIFY3(error!=0) failed (0 != 0)
```

and and I think this would be better.

```
VERIFY3(error != 0) failed (0 != 0)
```